src/instructlab/data/ingest_docs.py

```diff
@@ -0,0 +1,116 @@
+# Standard
+from pathlib import Path
+import glob
+import logging
+import os
+
+# Third Party
+from haystack import Pipeline  # type: ignore
+from haystack.components.converters import TextFileToDocument  # type: ignore
+from haystack.components.embedders import (  # type: ignore
+    SentenceTransformersDocumentEmbedder,
+)
+from haystack.components.preprocessors import DocumentCleaner  # type: ignore
+from haystack.components.writers import DocumentWriter  # type: ignore
+from milvus_haystack import MilvusDocumentStore  # type: ignore
+
+# First Party
+from instructlab.haystack.docling_splitter import DoclingDocumentSplitter
+from instructlab.rag.rag_configuration import (
+    document_store_configuration,
+    embedder_configuration,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def ingest_docs(
+    input_dir: str,
+    document_store_config: document_store_configuration,
+    embedder_config: embedder_configuration,
+):
+    pipeline = _create_pipeline(
+        document_store_config=document_store_config,
+        embedder_config=embedder_config,
+    )
+    _connect_components(pipeline)
+    _ingest_docs(pipeline=pipeline, input_dir=input_dir)
+
+
+def _create_pipeline(
```

Comment:
if we could add docstrings to these, that would be great!

Reply:
only to "public" functions/methods?